### PR TITLE
feat(extension): shared wallets feature flags are now controlled via PostHog

### PR DESCRIPTION
**Pull request by AngelCastilloB**
Currently the Shared Wallet feature is behind a build time feature flag

We’re looking to make this feature part of the new opt-in beta program, which requires it to be upgraded to a runtime feature flag, controlled remotely via PostHog

# Checklist

- [x] JIRA - \<[link](https://input-output.atlassian.net/browse/LW-11515)>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Read the flag from posthog

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/7009/11046403510/index.html) for [ae98171d](https://github.com/input-output-hk/lace/pull/1448/commits/ae98171df4e5ab1ebd2314b0431bfdb8668176ed)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 0      | 4       | 0     | 36    | ✅     |

File: apps/browser-extension-wallet/src/providers/ExperimentsProvider/config.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
 import { ExperimentName, ExperimentsConfig, FallbackConfiguration } from './types';
 
-export const fallbackConfiguration: FallbackConfiguration = {
-  [ExperimentName.COMBINED_NAME_PASSWORD_ONBOARDING_SCREEN]: 'control',
+export const getDefaultFeatureFlags = (): FallbackConfiguration => ({
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/providers/ExperimentsProvider/config.ts:3]: We could probably avoid this if we follow https://posthog.com/product-engineers/feature-flag-best-practices#5-bootstrap-your-flags-to-make-them-available-immediately in the posthog init functon

Comment (AngelCastilloB) [reply apps/browser-extension-wallet/src/providers/ExperimentsProvider/config.ts:3]: This is used to initialize the record on the constructor, and also to set default values if communication with post-hog fails. I think is a good idea to initialize the record with sensible default values (even if posthog connection fails)

Comment (AngelCastilloB) [reply apps/browser-extension-wallet/src/providers/ExperimentsProvider/config.ts:3]: I just want to make sure the featureFlags object is properly initialized:

```
    this.featureFlags = {
      [Wallet.Cardano.NetworkMagics.Mainnet]: getDefaultFeatureFlags(),
      [Wallet.Cardano.NetworkMagics.Preprod]: getDefaultFeatureFlags(),
      [Wallet.Cardano.NetworkMagics.Preview]: getDefaultFeatureFlags(),
      [Wallet.Cardano.NetworkMagics.Sanchonet]: getDefaultFeatureFlags()
    };
```
This object is a combination of the feature flags + the interpretation of the payload, I will look if there is an easy way to bootstrap this (including payload)

**Review by mchappell — APPROVED** [review]

File: apps/browser-extension-wallet/.env.example [hunk 2]
```diff
@@ -59,10 +59,7 @@ POSTHOG_EXCLUDED_EVENTS=
 POSTHOG_PRODUCTION_TOKEN_MAINNET=production-mainnet-token
 POSTHOG_PRODUCTION_TOKEN_PREPROD=production-preprod-token
 POSTHOG_PRODUCTION_TOKEN_PREVIEW=production-preview-token
-POSTHOG_DEV_TOKEN_MAINNET=dev-mainnet-token
-POSTHOG_DEV_TOKEN_PREPROD=dev-preprod-token
-POSTHOG_DEV_TOKEN_PREVIEW=dev-preview-token
-POSTHOG_DEV_TOKEN_SANCHONET=dev-sanchonet-token
+POSTHOG_DEV_TOKEN=phc_gH96Lx5lEVXTTWEyytSdTFPDk3Xsxwi4BqG88mKObd1
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/.env.example:62]:
```suggestion
POSTHOG_DEV_TOKEN='dev-token'
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/.env.example (hunk 2 above)
Comment (AngelCastilloB) [reply apps/browser-extension-wallet/.env.example]: done here https://github.com/input-output-hk/lace/pull/1448/commits/47dcca4a08102436f9442ff4838c21cb7328873a

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**